src/extension.ts

```diff
@@ -1737,8 +1747,34 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
```

Comment:
in the spirit of breaking down extension.ts, move this into it's own file under `src/components/kubectl/namespace.ts`

Reply:
yes done, and there are some access changes to extension.ts to allow access of them from other refactored files. Ideally I can access `kubectl` now in events file as well rather than having it as method  param - but that change will introduce a conflict will leave it for now as it is .WDYT?

Reply:
happy to leave it as is, and we can come back to it later.

Reply:
yes. thats what i did. as of now its own file and in future we can start working on the refactoring of each logical resource as separate PR for easy fixing them

Reply:
for constants we can export it rather doing DI as that will have shorter method param list and easier to read and maintain.

Reply:
@kameshsampath let's stick with the DI for now, since that's how many other components expect it to be used:

https://github.com/Azure/vscode-kubernetes-tools/blob/master/src/extension.ts#L151
https://github.com/Azure/vscode-kubernetes-tools/blob/master/src/extension.ts#L155
https://github.com/Azure/vscode-kubernetes-tools/blob/master/src/extension.ts#L157
https://github.com/Azure/vscode-kubernetes-tools/blob/master/src/extension.ts#L158

Reply:
Agreed! pushed new changes with DI pattern